dynamic/clientset/dynamic_clientset.go

```diff
@@ -33,75 +33,72 @@ import (
 type Clientset struct {
 	config    rest.Config
 	resources *dynamicdiscovery.ResourceMap
+	dc        dynamic.Interface
 }
 
-func New(config *rest.Config, resources *dynamicdiscovery.ResourceMap) *Clientset {
+func New(config *rest.Config, resources *dynamicdiscovery.ResourceMap) (*Clientset, error) {
+	// TODO: find a way to obtain APIResource for getting GroupVersion.
```

Comment:
@enisoc I shifted `dc` field from `ResourceClient` to `Clientset`. Not sure whether this is what you suggested. Feel free to sync by person if I get you wrong.

When calling `NewForConfig` within `New`, I couldn't find any way to obtain `apiVersion` and `resource` info, so I couldn't call `resources.Get(apiVersion, resource)` to obtain `the GroupVersion` for populating `config`.

Reply:
Thanks for verifying that I am on the right track. I will post log errors later if I could not resolve it. :)